api.planx.uk/modules/flows/validate/service.ts

```diff
@@ -206,4 +212,69 @@ const inviteToPayEnabled = (flowGraph: FlowGraph): boolean => {
   );
 };
 
+const validateFileTypes = (flowGraph: FlowGraph): ValidationResponse => {
+  // Get all passport variables set by FileUpload and/or FileUploadAndLabel
+  const allFileFns = [
+    ...getFileUploadNodeFns(flowGraph),
+    ...getFileUploadAndLabelNodeFns(flowGraph),
+  ];
+  if (allFileFns.length < 1) {
+    return {
+      title: "File types",
+      status: "Not applicable",
+      message: "Your flow is not using FileUpload or UploadAndLabel",
+    };
+  }
+
+  // Get all file types supported by current release of ODP Schema & compare
+  const validFileTypes = getValidSchemaValues("FileType");
+  const invalidFileFns: string[] = [];
+  allFileFns.forEach((fn) => {
+    if (!validFileTypes?.includes(fn)) {
+      invalidFileFns.push(fn);
+    }
+  });
+  if (invalidFileFns.length > 0) {
+    // Get unique fns with count of occurances
+    const countInvalidFileFns = countBy(invalidFileFns);
+    const summarisedInvalidFileFns: string[] = [];
+    Object.entries(countInvalidFileFns).map(([k, v]: [string, number]) => {
+      summarisedInvalidFileFns.push(`${k} (${v})`);
+    });
+    return {
+      title: "File types",
+      status: "Warn",
+      message: `Your FileUpload or UploadAndLabel are setting data fields that are not supported by the current release of the ODP Schema: ${summarisedInvalidFileFns.join(", ")}`,
+    };
+  }
+
+  return {
+    title: "File types",
+    status: "Pass",
+    message:
+      "Files collected via FileUpload or UploadAndLabel are all supported by the ODP Schema",
+  };
+};
+
+const getFileUploadNodeFns = (flowGraph: FlowGraph): string[] => {
+  const fileUploadNodes = Object.entries(flowGraph).filter(
+    (entry): entry is [string, Node] =>
```

Comment:
So, as of TS 5.5 I think we can lose the `entry is [string, Node]` here and it _should_ be inferred from `isComponentType()`.

Docs: https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#inferred-type-predicates

Reply:
Yep, you're correct this is just properly inferred from `isComponentType()` - removed unnecessary verbose-ness!